src/pybind/mgr/mgr_module.py

```diff
@@ -310,10 +310,21 @@ def __call__(self, func):
 
     def call(self, mgr, cmd_dict, inbuf):
         kwargs = {}
+        kwargs_switch = False
         for a, d in self.args_dict.items():
             if 'req' in d and d['req'] == "false" and a not in cmd_dict:
                 continue
-            kwargs[a.replace("-", "_")] = cmd_dict[a]
+            if kwargs_switch or (isinstance(cmd_dict[a], str) and '=' in cmd_dict[a]):
+                mgr.log.debug('found kwarg, assuming all following args are kw style')
+                kwargs_switch = True
+                try:
+                    k, arg = cmd_dict[a].split('=')
```

Comment:
```
2021-04-11T13:56:43.869 DEBUG:teuthology.orchestra.run.gibba003:> sudo /home/ubuntu/cephtest/cephadm --image quay.ceph.io/ceph-ci/ceph:33f0df555080d13dfd4cdf079a3f974d3b813ff4 shell -c /etc/ceph/ceph.conf -k /etc/ceph/ceph.client.admin.keyring --fsid 6b93e0fc-9acd-11eb-81
a3-001a4aab830c -- ceph orch daemon add mon 'gibba003:[v2:172.21.2.103:3301,v1:172.21.2.103:6790]=c'
2021-04-11T13:56:45.795 INFO:teuthology.orchestra.run.gibba003.stderr:Error EINVAL: Traceback (most recent call last):
2021-04-11T13:56:45.796 INFO:teuthology.orchestra.run.gibba003.stderr:  File "/usr/share/ceph/mgr/mgr_module.py", line 1223, in _handle_command
2021-04-11T13:56:45.796 INFO:teuthology.orchestra.run.gibba003.stderr:    return self.handle_command(inbuf, cmd)
2021-04-11T13:56:45.796 INFO:teuthology.orchestra.run.gibba003.stderr:  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 140, in handle_command
2021-04-11T13:56:45.796 INFO:teuthology.orchestra.run.gibba003.stderr:    return dispatch[cmd['prefix']].call(self, cmd, inbuf)
2021-04-11T13:56:45.797 INFO:teuthology.orchestra.run.gibba003.stderr:  File "/usr/share/ceph/mgr/mgr_module.py", line 331, in call
2021-04-11T13:56:45.797 INFO:teuthology.orchestra.run.gibba003.stderr:    return self.func(mgr, **kwargs)
2021-04-11T13:56:45.797 INFO:teuthology.orchestra.run.gibba003.stderr:  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 102, in <lambda>
2021-04-11T13:56:45.797 INFO:teuthology.orchestra.run.gibba003.stderr:    wrapper_copy = lambda *l_args, **l_kwargs: wrapper(*l_args, **l_kwargs)
2021-04-11T13:56:45.798 INFO:teuthology.orchestra.run.gibba003.stderr:  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 91, in wrapper
2021-04-11T13:56:45.798 INFO:teuthology.orchestra.run.gibba003.stderr:    return func(*args, **kwargs)
2021-04-11T13:56:45.798 INFO:teuthology.orchestra.run.gibba003.stderr:TypeError: _daemon_add_misc() got an unexpected keyword argument 'gibba003:[v2:172.21.2.103:3301,v1:172.21.2.103:6790]'
2021-04-11T13:56:45.798 INFO:teuthology.orchestra.run.gibba003.stderr:
2021-04-11T13:56:46.267 DEBUG:teuthology.orchestra.run:got remote process result: 22
```

i think this function somehow took `'gibba003:[v2:172.21.2.103:3301,v1:172.21.2.103:6790]=c'` as a kwarg, and used `"gibba003:[v2:172.21.2.103:3301,v1:172.21.2.103:6790]"` as the key.

see https://pulpito.ceph.com/kchai-2021-04-11_12:37:49-rados-wip-kefu-octopus-testing-2021-04-11-1851-distro-basic-gibba/6036056/

probably we should backport 922f962ae123b94ee7f92722190d19280f4ef6e3 as well.

Reply:
added